api/src/main/java/io/minio/MinioClient.java

```diff
@@ -4414,6 +4414,7 @@ private void putObject(String bucketName, String objectName, Long size, Object d
       // All parts have been uploaded, complete the multipart upload.
       completeMultipart(bucketName, objectName, uploadId, totalParts);
     } catch (RuntimeException e) {
+      abortMultipartUpload(bucketName, objectName, uploadId);
```

Comment:
Does `catch (Exception e)` catch `RuntimeException`?  If yes, you would need to remove `catch (RuntimeException e)`.

Reply:
I thought it did, but I assumed they were caught separately for a reason.  I'll re-test with one catch block and update this if it does the same thing.

Reply:
Ah, when building this without the `catch (RuntimeException e)` findbugs raises a `REC_CATCH_EXCEPTION` error:

> REC_CATCH_EXCEPTION: Exception is caught when Exception is not thrown
This method uses a try-catch block that catches Exception objects, but Exception is not thrown within the try block, and RuntimeException is not explicitly caught. It is a common bug pattern to say try { ... } catch (Exception e) { something } as a shorthand for catching a number of types of exception each of whose catch blocks is identical, but this construct also accidentally catches RuntimeException as well, masking potential bugs.
>
> A better approach is to either explicitly catch the specific exceptions that are thrown, or to explicitly catch RuntimeException exception, rethrow it, and then catch all non-Runtime Exceptions, as shown below:
>
>  try {
>    ...
>  } catch (RuntimeException e) {
>    throw e;
>  } catch (Exception e) {
>    ... deal with all non-runtime exceptions ...
>  }

I suspect this is why they were both caught before.  I could remove the `catch (Exception e)` but I'm not sure what that would let through.

Reply:
Removing `catch (Exception e)` is not correct. But removing `catch (RuntimeException e)` is more appropriate, but I don't know why it was added. Its better to leave as is.